lib/modules/asset/AssetService.ts

```diff
@@ -598,6 +600,82 @@ export class AssetService extends DigitalTwinService {
     return replacedAssets;
   }
 
+  /**
+   * Retrieve locales with the specified model and search all assets related to the model to update the locales.
+   * The operation is historized.
+   * @param request
+   * @param engineGroup
+   * @param model
+   */
+  public async updateModelLocales(
+    request: KuzzleRequest,
+    engineGroup: string,
+    model: string,
+  ): Promise<void> {
+    const { result } = await this.sdk.query({
+      action: "getAsset",
+      body: {},
+      controller: "device-manager/models",
+      engineGroup,
+      model,
+    });
+
+    const locales = result._source.asset.locales;
+
+    const engines = await ask<AskEngineList>("ask:device-manager:engine:list", {
+      group: engineGroup,
+    });
+
+    const targets = engines.map((engine) => ({
+      collections: [InternalCollection.ASSETS],
+      index: engine.index,
+    }));
+
+    const assets = await this.sdk.query<
```

Comment:
This request will only return 10 results... you should :
* add a scoll ID to be able to iter on more that 10000 results
* use a SearchResult and the .next() method to iterate over all the results
* set a size to work on large data chunk to avoid doing to many requests